github-runner-manager/tests/unit/test_github_client.py

```diff
@@ -616,3 +629,30 @@ def test_get_runner_not_found(
     )
     with pytest.raises(GithubRunnerNotFoundError):
         _ = github_client.get_runner(path, prefix, runner_id)
+
+
+def test_delete_runner_busy(
+    github_client: GithubClient,
+    monkeypatch: pytest.MonkeyPatch,
+):
+    """
+    arrange: A mocked GhApi Github Client that raises 404 when a runner is requested.
+    act: Call get_runner in GithubClient.
+    assert: The exception GithubRunnerNotFoundError is raised.
+    """
+    path = GitHubOrg(org=secrets.token_hex(16), group=secrets.token_hex(16))
+    runner_id = 1
+
+    error_body = """
+        {
+        "message": "Bad request - Runner test-bs69mhr2-0-n-5dbb110595a9 is currently running a job and cannot be deleted.",
+        "documentation_url": "https://docs.github.com/rest/actions/self-hosted-runners#delete-a-self-hosted-runner-from-a-repository",
+        "status": "422"
+        }
```

Comment:
Thats nice. It serves as our documentation (because github is not providing it).